src/ui/component/modal/modal.tsx

```diff
@@ -0,0 +1,90 @@
+import { type ReactNode, type FC, useCallback, useRef, useEffect } from 'react'
+import { createPortal } from 'react-dom'
+import classNames from 'classnames'
+import { CSSTransition } from 'react-transition-group'
+import { useOnKeyboard } from '@/ui/hook/useOnKeyboard'
+import './modal.scss'
+
+type ModalProps = {
+  isOpen: boolean
+  onClose: () => void
+  closeOnEsc?: boolean
+  motionPreset?: 'zoom' | 'none'
+  isCentered?: boolean
+  children?: ReactNode
+  classes?: {
+    main?: string
+    overlay?: string
+  }
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const Modal: FC<ModalProps> = ({
+  isOpen,
+  onClose,
+  closeOnEsc = true,
+  motionPreset = 'zoom',
+  isCentered = true,
+  children,
+  classes
+}) => {
+  const containerRef = useRef<HTMLElement | null>(null)
+
+  useEffect(() => {
+    containerRef.current = document.getElementById('modal-root')
+  }, [])
+
+  const handleEscape = useCallback(
+    (event: KeyboardEvent) => {
+      if (closeOnEsc && event.key === 'Escape') {
+        onClose()
+      }
+    },
+    [onClose, closeOnEsc]
+  )
+
+  useOnKeyboard(handleEscape)
+
+  const handleOverlayClick = useCallback(() => {
+    onClose()
+  }, [onClose])
+
+  const handleDialogOnClick = useCallback((e: React.MouseEvent): void => {
```

Comment:
If I understand this correctly, you made this function to prevent modal closing when clicking on it.
If it's your intention, I suggest you to rename this function to explicitely declare its purpose.